flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -461,8 +463,9 @@ public LocatableInputSplitAssigner getInputSplitAssigner(FileInputSplit[] splits
 		
 		// take the desired number of splits into account
 		minNumSplits = Math.max(minNumSplits, this.numSplits);
-		
-		final Path path = this.filePath;
+
+		final Path path = new Path(URLDecoder.decode(this.filePath.toString(), Charset.defaultCharset().name()));
```

Comment:
We can also use ```parentDir.getAbsoluteFile().toString()``` to hack that. Not very sure about which way is better. I would think ```parentDir.getAbsoluteFile().toString()``` is better as it can obtain the whole path. What do you think ? @zentol